main/modes/jumper/mode_jumper.c

```diff
@@ -1803,27 +1809,59 @@ void jumperDrawEffects(display_t* d)
                 20 + j->scene->blockOffset_y, false, false, 0);
 }
 
-void jumperDrawHud(display_t* d, font_t* prompt, font_t* font, font_t* outline)
+void jumperDrawHud(display_t* d, font_t* prompt, font_t* font, font_t* outline, font_t* smaller_prompt)
 {
     char textBuffer[12];
+    char textBuffer2[12];
     snprintf(textBuffer, sizeof(textBuffer) - 1, "LVL %d", j->scene->level);
 
     if (j->scene->level < 10)
     {
-        drawText(d, font, c555, textBuffer, 180, 220);
+        drawText(d, font, c555, textBuffer, 190, 220);
     }
     else
     {
-        drawText(d, font, c555, textBuffer, 164, 220);
+        drawText(d, font, c555, textBuffer, 174, 220);
     }
 
-    drawText(d, prompt, c555, "SCORE", 28, 10);
     snprintf(textBuffer, sizeof(textBuffer) - 1, "%u", j->scene->score);
-    drawText(d, prompt, c555, textBuffer, 28, 30);
+    snprintf(textBuffer2, sizeof(textBuffer2) - 1, "%u", j->highScore);
+
+    font_t* scoreFont;
+    int16_t scoreWidth;
+    int16_t hiScoreWidth;
+    int16_t maxScoreWidth;
+
+    if(!j->scene->smallerScoreFont)
+    {
+        scoreFont = prompt;
+        scoreWidth = textWidth(prompt, textBuffer);
+        hiScoreWidth = textWidth(prompt, textBuffer2);
+        maxScoreWidth = textWidth(prompt, "222222222");
+    }
+
+    if(j->scene->smallerScoreFont || scoreWidth > maxScoreWidth || hiScoreWidth > maxScoreWidth)
+    {
+        j->scene->smallerScoreFont = true;
+        scoreFont = smaller_prompt;
+        scoreWidth = textWidth(smaller_prompt, textBuffer);
+        hiScoreWidth = textWidth(smaller_prompt, textBuffer2);
+        maxScoreWidth = textWidth(smaller_prompt, "2122222222");
```

Comment:
`2122222222` not `2222222222`?

Reply:
the max `int32_t` that can be pulled from NVS seems to be 2,147,483,648, despite the max for an `int32_t` being 1 less at 2,147,483,647. on further inspection, the score can continue to increase past this value since the game struct stores it as a `uint32_t`, but predictable overflow errors ensue. I'm leaning towards changing the struct's score variable type to `int32_t` to match NVS, in which case the second-most-significant digit would never reach 2

Reply:
Immediately upon typing and sending that, I realized that a 0 is the same width as a 2, so you're correct